File: src/Java/201-300/263.IsUglyNumber.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.Comparator;
import java.util.Queue;
import java.util.Set;
import java.util.HashSet;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.Map;

class Solution {
    public boolean isUgly(int num) {
        if (num <= 0) 
            return false;
        while (num % 2 == 0) 
            num /= 2;
        while (num % 3 == 0) 
            num /= 3;
        while (num % 5 == 0) 
            num /= 5;
        return num == 1;
    }
}

File: src/Python/201-300/263.IsUglyNumber.py

class Solution:
    def isUgly(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num <= 0:
            return False
        while num % 2 == 0:
            num = num // 2
        while num % 3 == 0:
            num = num // 3
        while num % 5 == 0:
            num = num // 5
        if num == 1:
            return True
        return False

if __name__ == '__main__':
    solution = Solution()
    print(solution.isUgly(1));
    print(solution.isUgly(2));
    print(solution.isUgly(8));
    print(solution.isUgly(12));
    print(solution.isUgly(14));
    print(solution.isUgly(123));
else:
    pass
